python/GafferUI/NodeGraph.py

```diff
@@ -351,6 +356,18 @@ def __frame( self, nodes, extend = False ) :
 
 		self.__gadgetWidget.getViewportGadget().frame( bound )
 
+	def __nodesHaveUIPosition( self, scriptNode ) :
+
+		for node in scriptNode.children( Gaffer.Node ) :
+			if '__uiPosition' in node :
+				return True
+
+		return False
+
+	def __layoutAllNodes( self , graphGadget, scriptNode ) :
+
+		graphGadget.getLayout().layoutNodes( graphGadget, Gaffer.StandardSet( scriptNode.children( Gaffer.Node ) ) )
```

Comment:
Bear in mind that the NodeGraph isn't necessarily viewing the nodes parented to the script - it could be viewing nodes inside a box, or any other parent. That's actually one of the biggest use cases for your auto-layout that we have here at IE, since our auto generated networks are generally internal to a box. I think we need to use `graphGadget.getRoot()` here, and also make sure we're doing the layout if necessary in `__rootChanged()`.